RecoEgamma/EgammaTools/interface/MVAValueMapProducer.h

```diff
@@ -74,32 +66,56 @@ MVAValueMapProducer<ParticleType>::MVAValueMapProducer(const edm::ParameterSet&
   srcMiniAOD_ = mayConsume<edm::View<ParticleType> >(iConfig.getParameter<edm::InputTag>("srcMiniAOD"));
 
   // Loop over the list of MVA configurations passed here from python and
-  // construct all requested MVA estimators.
-  const auto& all_mvas = mva_cache->allMVAs();
-  for( auto mvaItr = all_mvas.begin(); mvaItr != all_mvas.end(); ++mvaItr ) {
-    // set the consumes
-    mvaItr->second->setConsumes(consumesCollector());
+  // construct all requested MVA esimtators.
+  const std::vector<edm::ParameterSet>& mvaEstimatorConfigs
+    = iConfig.getParameterSetVector("mvaConfigurations");
+
+  for( auto &imva : mvaEstimatorConfigs ){
+
+    std::unique_ptr<AnyMVAEstimatorRun2Base> thisEstimator;
+    thisEstimator.reset(NULL);
+    if( !imva.empty() ) {
+      const std::string& pName = imva.getParameter<std::string>("mvaName");
+      // The factory below constructs the MVA of the appropriate type based
+      // on the "mvaName" which is the name of the derived MVA class (plugin)
+      AnyMVAEstimatorRun2Base *estimator = AnyMVAEstimatorRun2Factory::get()->create(pName, imva);
```

Comment:
Change this to 
```
mvaEstimators_.emplace_back( std::make_unique<AnyMVAEstimatorRun2Base>(AnyMVAEstimatorRun2Factory::get()->create(pName, imva)) );
```

Reply:
Unfortunately it doesn't work like that...
```
error: invalid new-expression of abstract class type 'AnyMVAEstimatorRun2Base'
```
I'm trying find a solution.

Reply:
Yeah, got that wrong. Try doing
```
mvaEstimators_.emplace_back(AnyMVAEstimatorRun2Factory::get()->create(pName, imva));
```

Reply:
Thanks! Got to read what this `emplace` does after the commit...